components/transactions/TransactionRejectButton.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { FormattedMessage } from 'react-intl';
+
+import { getErrorFromGraphqlException } from '../../lib/errors/';
+import { GraphQLContext } from '../../lib/graphql/context';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+
+import { Box, Flex } from '../Grid';
+import MessageBox from '../MessageBox';
+import StyledButton from '../StyledButton';
+
+/* eslint-disable graphql/template-strings */
+export const rejectTransactionMutation = gqlV2/* GraphQL */ `
+  mutation RejectTransaction($transaction: TransactionReferenceInput!) {
+    rejectTransaction(transaction: $transaction) {
+      id
+    }
+  }
+`;
+
+const TransactionRejectButton = props => {
+  const [rejectTransaction, { loading, error }] = useMutation(rejectTransactionMutation, {
+    context: API_V2_CONTEXT,
+  });
+  const transactionsDataContext = React.useContext(GraphQLContext);
```

Comment:
Not a requested change:

I understand that this is the same logic than `TransactionRefundButton`, but I'm not a fan of using context for this. If we add this button on another page that doesn't use a GraphQL context we won't be able to refetch the data.

I think passing down a `refetchData` prop is a better fit for that kind of use cases. Ping @kewitz

Reply:
Any input on this? Would be good to know as I'm still not getting the immediately updated transaction.

Reply:
What would you want to see updated, the order status? Did you check if the API properly returns the updated order?

Reply:
Agreed. Preferably with a generic name like "onSuccess".

Reply:
Added this to both `TransactionRejectButton` and `TransactionRefundButton` 🎉